src/main/java/com/stuypulse/robot/subsystems/Edwin.java

```diff
@@ -0,0 +1,229 @@
+            new CANSparkMax(Ports.Edwin.RIGHT_BOTTOM, MotorType.kBrushless)
+        );
+
+        drivetrain = new DifferentialDrive(left, right);
+
+        leftGreyhill = new Encoder(Ports.Edwin.Encoders.LEFT_A, Ports.Edwin.Encoders.LEFT_B);
+        rightGreyhill = new Encoder(Ports.Edwin.Encoders.RIGHT_A, Ports.Edwin.Encoders.RIGHT_B);
+        
+        setGreyhillDistancePerPulse(EdwinSettings.Encoders.GREYHILL_DISTANCE_PER_PULSE);
+
+        leftController = new Feedforward.Drivetrain(Motion.Feedforward.kS, Motion.Feedforward.kV, Motion.Feedforward.kA).velocity()
+            .add(new PIDController(Motion.PID.kP, Motion.PID.kI, Motion.PID.kD));
+        
+        rightController = new Feedforward.Drivetrain(Motion.Feedforward.kS, Motion.Feedforward.kV, Motion.Feedforward.kA).velocity()
+            .add(new PIDController(Motion.PID.kP, Motion.PID.kI, Motion.PID.kD));
+
+        leftTargetSpeed = 0;
+        rightTargetSpeed = 0;
+
+        odometry = new DifferentialDriveOdometry(getRotation2d());
+        kinematics = new DifferentialDriveKinematics(EdwinSettings.TRACK_WIDTH);
+        navx = new AHRS(SPI.Port.kMXP);
+
+        right.setInverted(true);
+        reset(EdwinSettings.Odometry.STARTING_POSITION);
+
+        field = new Field2d();
+    }
+
+    /*********************
+     * ENCODER FUNCTIONS *
+     *********************/
+
+    private void setGreyhillDistancePerPulse(double distance) {
+        rightGreyhill.setDistancePerPulse(distance);
+        rightGreyhill.reset();
+
+        leftGreyhill.setDistancePerPulse(distance);
+        leftGreyhill.reset();
+    }
+
+    // Distance
+    private double getLeftDistance() {
+        return leftGreyhill.getDistance();
+    }
+
+    private double getRightDistance() {
+        return rightGreyhill.getDistance();
+    }
+
+    private double getDistance() {
+        return (getLeftDistance() + getRightDistance()) / 2.0;
+    }
+
+    // Velocity
+    private double getLeftVelocity() {
+        return leftGreyhill.getRate();
+    }
+
+    private double getRightVelocity() {
+        return rightGreyhill.getRate();
+    }
+
+    private double getVelocity() {
+        return (getLeftVelocity() + getRightVelocity()) / 2.0;
+    }
+
+    /***************
+     * ROBOT ANGLE *
+     ***************/
+
+    // Gets current Angle of the Robot as a double (contiuous / not +-180)
+    public double getRawGyroAngle() {
+        return navx.getAngle();
+    }
+
+    // Gets current Angle of the Robot
+    public Angle getGyroAngle() {
+        return Angle.fromDegrees(getRawGyroAngle());
+    }
+
+    /**********************
+     * ODOMETRY FUNCTIONS *
+     **********************/
+
+    private void updateOdometry() {
+        odometry.update(getRotation2d(), getLeftDistance(), getRightDistance());
+    }
+
+    public Rotation2d getRotation2d() {
+        // TODO: check if this needs to be negative
+        return getGyroAngle().negative().getRotation2d();
```

Comment:
gyro.getRotation2d() ... many of the frc gyros impleent the gyro interface which handles this coordinate plane conversion for you in the getRotation2d() method. 

also rotations2d's do not loop the angle around, so we don't lose continuity as long as no  operations are done with the rotation2d object

 (e..g getRotation2d().getDegrees() can return 13209 degrees cause it wont loop it around yet, whcih is good) 

(once i do getRotation2d().plus(new Rotation2d(7)) is when the output is guaranteed to be between -pi and +pi because it uses sin/cos which loop ur angle value)